lib/jxl/pack_signed.h

```diff
@@ -29,7 +29,8 @@ constexpr uint32_t PackSigned(int32_t value)
 // convenience its type is `size_t` (i.e. `uint32_t` or `uint64_t`).
 constexpr int32_t UnpackSigned(size_t value)
     JXL_NO_SANITIZE("unsigned-integer-overflow") {
-  JXL_DASSERT((value & 0xFFFFFFFF) == value);  // no-op in 32-bit build
+  // TODO(Ivan): fails in C++11 mode, restore with a guard?
```

Comment:
When does this fail? This function is not supposed to be called with an input that exceeds uint32...

Reply:
Yes, it fails to compile in C++11 mode on Clang and GCC insisting that `__builtin_trap()` is not `constexpr`.

Reply:
```
[build] In file included from /home/ivan/libjxl/lib/jxl/entropy_coder.cc:21:
[build] /home/ivan/libjxl/lib/jxl/pack_signed.h: In function ‘constexpr int32_t jxl::UnpackSigned(size_t)’:
[build] /home/ivan/libjxl/lib/jxl/pack_signed.h:34:1: error: body of ‘constexpr’ function ‘constexpr int32_t jxl::UnpackSigned(size_t)’ not a return-statement
[build]    34 | }
[build]       | ^
```

Reply:
Oh, I see.